server/src/http_login.go

```diff
@@ -131,6 +132,35 @@ func httpLogin(c *gin.Context) {
 				http.Error(w, "That is not the correct password.", http.StatusUnauthorized)
 				return
 			}
+
+			// If they have provided a new password, replace the old one
+			if data.NewPassword != "" {
```

Comment:
If NewPassword is not passed, does it default to `""` or `nil`?

Reply:
It is empty string